Topic 1/src/point6/Transaction.java

```diff
@@ -0,0 +1,46 @@
+package point6;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Transaction {
+	private static final Transaction _INSTANCE = new Transaction();
+	
+	private List<ShoppingCart> transactions;
+	private MailList mailList;
+	
+	private Transaction(){
+		this.transactions= new ArrayList<ShoppingCart>();
+	}
+
+	public static Transaction getTransaction(){
+		return _INSTANCE;
+	}
+	
+	public int numberTransactions(){
+		return this.transactions.size();
+	}
+	
+	public void addTransaction(ShoppingCart shoppingCart){
```

Comment:
[High] You don't need to notify about transaction here. You can do it when the user pays the shopping cart.